### ISSUE TITLE
OpenMailAppPlugin.kt: (25, 9): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin.

### ISSUE THREAD

**Issue by nerder** [HomeXLabs/open-mail-app-flutter#37]
Since updating `flutter` to the latest available version in the channel `beta` we are facing an issue while trying to compile.

*flutter doctor -v*
```
[✓] Flutter (Channel beta, 2.9.0-0.1.pre, on macOS 11.4 20F71 darwin-x64, locale en-ES)
    • Flutter version 2.9.0-0.1.pre at /Users/nerder/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 8f1f9c10f0 (18 hours ago), 2021-12-14 13:41:48 -0800
    • Engine revision 234aca678a
    • Dart version 2.16.0 (build 2.16.0-80.1.beta)
    • DevTools version 2.9.1
```

```
e: Incompatible classes were found in dependencies. Remove them from the classpath or use '-Xskip-metadata-version-check' to suppress errors
e: /Users/nerder/.gradle/caches/transforms-2/files-2.1/121c56a4eb3b9888b3a925c5a1b06c22/jetified-kotlinx-coroutines-android-1.5.2.jar!/META-INF/kotlinx-coroutines-android.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
e: /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/META-INF/kotlin-stdlib.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
e: /Users/nerder/.gradle/caches/transforms-2/files-2.1/86ffb183faa2bd338bfefd08560447d3/jetified-kotlin-stdlib-common-1.5.31.jar!/META-INF/kotlin-stdlib-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
e: /Users/nerder/.gradle/caches/transforms-2/files-2.1/8cd76d3c199ed7f84af27a672071d60c/jetified-kotlin-stdlib-jdk8-1.5.30.jar!/META-INF/kotlin-stdlib-jdk8.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
e: /Users/nerder/.gradle/caches/transforms-2/files-2.1/9cf3398f5e61cc366c5ee0aa5c3bc390/jetified-kotlin-stdlib-jdk7-1.5.30.jar!/META-INF/kotlin-stdlib-jdk7.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
e: /Users/nerder/.gradle/caches/transforms-2/files-2.1/9d2040e319b51dc246c400f1b2db4972/jetified-window-1.0.0-beta04-api.jar!/META-INF/window_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
e: /Users/nerder/.gradle/caches/transforms-2/files-2.1/ab2037cf9b0bc48dddc9452783f6bf2f/jetified-kotlinx-coroutines-core-jvm-1.5.2.jar!/META-INF/kotlinx-coroutines-core.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
e: /Users/nerder/.gradle/caches/transforms-2/files-2.1/b0053ae52ec47f3a01bcb7033b2bbe63/jetified-window-java-1.0.0-beta04-api.jar!/META-INF/window-java_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (25, 9): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (26, 17): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (27, 9): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (40, 10): Class 'kotlin.jvm.JvmStatic' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/jvm/JvmStatic.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (44, 21): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (45, 20): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (50, 9): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (54, 9): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (56, 20): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (57, 16): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (59, 20): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (60, 16): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (62, 20): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (63, 16): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (65, 20): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (66, 16): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (69, 20): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (71, 20): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (76, 17): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (84, 38): Unresolved reference: isNotEmpty
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (86, 66): Unresolved reference: first
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (91, 37): Unresolved reference: mutableListOf
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (92, 25): Unresolved reference: until
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (95, 72): Unresolved reference: let
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (95, 78): Cannot infer a type for this parameter. Please specify it explicitly.
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (109, 32): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (118, 77): Unresolved reference: java
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (122, 38): Unresolved reference: isNotEmpty
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (123, 91): Unresolved reference: apply
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (124, 17): Unresolved reference: data
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (125, 17): Unresolved reference: type
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (126, 17): Unresolved reference: setClassName
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (126, 55): Unresolved reference: first
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (126, 114): Unresolved reference: first
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (128, 17): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (129, 17): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (130, 17): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (131, 17): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (132, 17): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (135, 41): Unresolved reference: mutableListOf
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (136, 25): Unresolved reference: until
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (141, 62): Unresolved reference: apply
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (142, 37): Unresolved reference: data
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (143, 37): Unresolved reference: type
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (144, 37): Unresolved reference: setClassName
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (145, 37): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (146, 37): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (147, 37): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (148, 37): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (149, 37): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (161, 32): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (173, 62): Unresolved reference: firstOrNull
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (174, 13): Unresolved reference: it
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (181, 26): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (182, 28): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (188, 77): Unresolved reference: java
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (192, 62): Unresolved reference: firstOrNull
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (193, 13): Unresolved reference: it
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (196, 63): Unresolved reference: apply
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (197, 13): Unresolved reference: data
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (198, 13): Unresolved reference: type
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (199, 13): Unresolved reference: setClassName
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (200, 13): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (201, 13): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (202, 13): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (203, 13): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (204, 13): Unresolved reference: putExtra
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (208, 28): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
The class is loaded from /Users/nerder/.gradle/caches/transforms-2/files-2.1/246d7067bcc442dabda444cac586ac0f/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (218, 45): Unresolved reference: isNotEmpty
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (219, 28): Unresolved reference: mutableListOf
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (220, 25): Unresolved reference: until
e: /Users/nerder/.pub-cache/hosted/pub.dartlang.org/open_mail_app-0.3.0/android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt: (226, 13): Unresolved reference: emptyList

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':open_mail_app:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2m 55s
[!] Your project requires a newer version of the Kotlin Gradle plugin.
    Find the latest version on https://kotlinlang.org/docs/gradle.html#plugin-and-versions, then update /Users/nerder/Developments/strive/checkin/android/build.gradle:
    ext.kotlin_version = '<latest-version>'
Exception: Gradle task assembleDevDebug failed with exit code 1
```

**Comment by nerder** [comment]
@MisterJimson Have you seen this by any chance?

**Comment by MisterJimson** [comment]
Should be fixed in 0.4.2. Note that I wasn't running into this issue personally.
